### PR TITLE
Merge HDRP staging

### PR DESCRIPTION
**Pull request by sebastienlagarde**
### Purpose of this PR
- Remove unused code in AdditionalShadowData.cs (#3665)
- Disable sky when in matcap view (#3667)
- Hdrp/fix wizard default scene prefab extension (#3538)
- Moving ray tracing parameters to Volume system, HDAddditionalLightData and fixing compilation errors (#3658)

### Release Notes
Please add any useful notes about the feature/fix that might be helpful for others.

---
### Testing status
**Katana Tests**: First off we need to make sure the Katana SRP tests are green?

**Manual Tests**: What did you do?
- [ ] Opened test project + Run graphic tests locally
- [ ] Built a player
- [ ] Checked new UI names with UX convention
- [ ] Tested UI multi-edition + Undo/Redo
- [ ] C# and shader warnings (supress shader cache to see them)
- [ ] Checked new resources path for the reloader (in devloper mode, you have a button at end of resources that check the pathes)
- Other: 

**Automated Tests**: What did you setup? (Add a screenshot or the reference image of the test please)

https://katana.bf.unity3d.com/projects/com.unity.render-pipelines/builders?ScriptableRenderLoop_branch=HDRP%2Fstaging&automation-tools_branch=master&unity_branch=trunk&sort=1-asc

---
### Overall Product Risks

---
### Comments to reviewers
All PR have already been reviewed

### CODE REVIEW

File: com.unity.render-pipelines.high-definition/Runtime/Lighting/LightLoop/LightLoop.cs [hunk 1]
```diff
@@ -95,6 +95,9 @@ public class LightDefinitions
         public static uint s_LightFeatureMaskFlagsOpaque = 0xFFF000 & ~((uint)LightFeatureFlags.SSRefraction); // Opaque don't support screen space refraction
         public static uint s_LightFeatureMaskFlagsTransparent = 0xFFF000 & ~((uint)LightFeatureFlags.SSReflection); // Transparent don't support screen space reflection
         public static uint s_MaterialFeatureMaskFlags = 0x000FFF;   // don't use all bits just to be safe from signed and/or float conversions :/
+
+        // Ray traced shadows
+        public static uint s_MaxRayTracedShadows = 4;
```
Comment (sebastienlagarde) [inline com.unity.render-pipelines.high-definition/Runtime/Lighting/LightLoop/LightLoop.cs]: This should be expose in HDRP asset

Comment (anisunity) [reply com.unity.render-pipelines.high-definition/Runtime/Lighting/LightLoop/LightLoop.cs]: Will happen with the refactor of the screen space shadows yes

File: com.unity.render-pipelines.high-definition/Runtime/Lighting/LightLoop/LightLoop.cs [hunk 2]
```diff
@@ -1222,10 +1234,11 @@ public bool GetDirectionalLightData(CommandBuffer cmd, GPULightType gpuLightType
              }
 
 #if ENABLE_RAYTRACING
-            if(gpuLightType == GPULightType.Rectangle && lightComponent.shadows != LightShadows.None && areaShadowIndex < maxAreaLightShadows && additionalLightData.useRayTracedShadows)
+            if(gpuLightType == GPULightType.Rectangle && lightComponent.shadows != LightShadows.None && areaShadowIndex < LightDefinitions.s_MaxRayTracedShadows && additionalLightData.useRayTracedShadows)
```
Comment (sebastienlagarde) [inline com.unity.render-pipelines.high-definition/Runtime/Lighting/LightLoop/LightLoop.cs]: I guess with have a mismatch between rasterized shadow and raytraced shadow here with areaShadowIndex
areaShadowIndex < LightDefinitions.s_MaxRayTracedShadows

File: com.unity.render-pipelines.high-definition/Runtime/Debug/RayCountManager.cs [hunk 3]
```diff
@@ -29,16 +29,16 @@ public enum RayCountValues
         
         // HDRP Resources
         DebugDisplaySettings m_DebugDisplaySettings;
-        RenderPipelineResources m_PipelineResources;
+        HDRenderPipelineRayTracingResources m_PipelineResources;
```
Comment (sebastienlagarde) [inline com.unity.render-pipelines.high-definition/Runtime/Debug/RayCountManager.cs]: Let's talk

File: com.unity.render-pipelines.high-definition/Runtime/Lighting/ScreenSpaceLighting/AmbientOcclusion.cs [hunk 4]
```diff
@@ -28,6 +43,9 @@ public class AmbientOcclusionSystem
         enum MipLevel { Original, L1, L2, L3, L4, L5, L6, Count }
 
         RenderPipelineResources m_Resources;
+#if ENABLE_RAYTRACING
+        HDRenderPipelineRayTracingResources m_RTResources;
```
Comment (sebastienlagarde) [inline com.unity.render-pipelines.high-definition/Runtime/Lighting/ScreenSpaceLighting/AmbientOcclusion.cs]: HDRenderPipelineRayTracingResources should be inside RenderPipelineResources

File: com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/HDRenderPipeline.cs [hunk 5]
```diff
@@ -385,6 +385,13 @@ void UpgradeResourcesIfNeeded()
                     = UnityEditor.AssetDatabase.LoadAssetAtPath<RenderPipelineResources>(HDUtils.GetHDRenderPipelinePath() + "Runtime/RenderPipelineResources/HDRenderPipelineResources.asset");
 			ResourceReloader.ReloadAllNullIn((GraphicsSettings.renderPipelineAsset as HDRenderPipelineAsset).renderPipelineResources, HDUtils.GetHDRenderPipelinePath());
 
+#if ENABLE_RAYTRACING
+            if ((GraphicsSettings.renderPipelineAsset as HDRenderPipelineAsset).renderPipelineRayTracingResources == null)
+                (GraphicsSettings.renderPipelineAsset as HDRenderPipelineAsset).renderPipelineRayTracingResources
+                    = UnityEditor.AssetDatabase.LoadAssetAtPath<HDRenderPipelineRayTracingResources>(HDUtils.GetHDRenderPipelinePath() + "Runtime/RenderPipelineResources/HDRenderPipelineRayTracingResources.asset");
```
Comment (sebastienlagarde) [inline com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/HDRenderPipeline.cs]: Do we need this ?

File: com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/HDRenderPipelineRayTracingResources.cs [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+        public ComputeShader lightClusterBuildCS;
+        [Reload("Runtime/RenderPipeline/Raytracing/Shaders/DebugLightCluster.compute")]
+        public ComputeShader lightClusterDebugCS;
+
+        // Indirect Diffuse
+#if ENABLE_RAYTRACING
+        [Reload("Runtime/RenderPipeline/Raytracing/Shaders/RaytracingIndirectDiffuse.raytrace")]
+        public RaytracingShader indirectDiffuseRaytracing;
+#endif
+        [Reload("Runtime/RenderPipeline/Raytracing/Shaders/IndirectDiffuseAccumulation.compute")]
```
Comment (sebastienlagarde) [inline com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/HDRenderPipelineRayTracingResources.cs]: Please include all raytrace only shader as well

File: com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/HDRenderPipelineRayTracingResources.cs [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+
+    #if UNITY_EDITOR
+        [UnityEditor.CustomEditor(typeof(HDRenderPipelineRayTracingResources))]
+        class RenderPipelineRayTracingResourcesEditor : UnityEditor.Editor
+        {
+            public override void OnInspectorGUI()
+            {
+                DrawDefaultInspector();
+
+                // Add a "Reload All" button in inspector when we are in developer's mode
```
Comment (sebastienlagarde) [inline com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/HDRenderPipelineRayTracingResources.cs]: Nice: Let's add sur on RenderPipeline Resources to refill everything, not just raytrace

File: com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/Raytracing/HDRaytracingAmbientOcclusion.cs [hunk 8]
```diff
@@ -127,81 +122,70 @@ public void RenderAO(HDCamera hdCamera, CommandBuffer cmd, RTHandleSystem.RTHand
...
-            cmd.SetRaytracingTextureParam(aoShader, HDShaderIDs._RaytracingHitDistanceTexture, m_HitDistanceBuffer);
             cmd.SetRaytracingTextureParam(aoShader, HDShaderIDs._RaytracingVSNormalTexture, m_ViewSpaceNormalBuffer);
 
             // Run the calculus
             cmd.DispatchRays(aoShader, m_RayGenShaderName, (uint)hdCamera.actualWidth, (uint)hdCamera.actualHeight, 1);
 
             using (new ProfilingSample(cmd, "Filter Reflection", CustomSamplerId.RaytracingAmbientOcclusion.GetSampler()))
             {
-                switch(rtEnvironement.aoFilterMode)
+                if(aoSettings.enableFilter.value)
```
Comment (sebastienlagarde) [inline com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/Raytracing/HDRaytracingAmbientOcclusion.cs]: "if (" (at several place :))

File: com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/Raytracing/HDRaytracingManager.cs [hunk 9]
```diff
@@ -314,21 +316,21 @@ public void UpdateFrameData()
             if (rtEnv == null) return;
 
             // If AO is on flag its RAS needUpdate
-            if (rtEnv.raytracedAO)
+            // if (rtEnv.raytracedAO)
```
Comment (sebastienlagarde) [inline com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/Raytracing/HDRaytracingManager.cs]: comment why it is commented :)

File: com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/Raytracing/HDRaytracingReflection.cs [hunk 10]
```diff
@@ -164,22 +171,155 @@ public void RenderReflections(HDCamera hdCamera, CommandBuffer cmd, RTHandleSyst
...
             // Compute the actual resolution that is needed base on the quality
-            uint widthResolution = 1, heightResolution = 1;
-            switch (rtEnvironement.reflQualityMode)
+            uint widthResolution = (uint)hdCamera.actualWidth / 2;
+            uint heightResolution = (uint)hdCamera.actualHeight / 2;
+
+            // Force to disable specular lighting
+            cmd.SetGlobalInt(HDShaderIDs._EnableSpecularLighting, 0);
+
+            // Run the calculus
```
Comment (sebastienlagarde) [inline com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/Raytracing/HDRaytracingReflection.cs]: "calculus" is not english

File: com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/Settings/RenderPipelineSettings.cs [hunk 11]
```diff
@@ -56,7 +62,9 @@ public enum ColorBufferFormat
             decalSettings = GlobalDecalSettings.@default,
             postProcessSettings = GlobalPostProcessSettings.@default,
             dynamicResolutionSettings = GlobalDynamicResolutionSettings.@default,
-            lowresTransparentSettings = GlobalLowResolutionTransparencySettings.@default
+            lowresTransparentSettings = GlobalLowResolutionTransparencySettings.@default,
+            supportRayTracing = false,
+            supportedRaytracingTier = RaytracingTier.Tier2,
```
Comment (sebastienlagarde) [inline com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/Settings/RenderPipelineSettings.cs]: Set this in a RaytracingSettings class

**Comment by sebastienlagarde** [comment]
Katana is green

**Pull request merged**